### ISSUE TITLE
[2021] R-Drop: Regularized Dropout for Neural Networks

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#135]
## 0. 論文

| タイトル | R-Drop: Regularized Dropout for Neural Networks |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2106.14448 https://github.com/dropreg/R-Drop |
| 著者    | Xiaobo Liang, Lijun Wu, Juntao Li, Yue Wang, Qi Meng, Tao Qin, Wei Chen, Min Zhang, Tie-Yan Liu |
| 投稿日付 | 2021/6/28 on arxiv |

## 1. どんなもの？

Dropout を発展させた正則化手法である R-Drop を提案。ある入力を異なる dropout mask で2回 model を通し、それを2つの sub model の出力と考えて、2つの出力が近づくように最終層での分布の KD Divergence をロスをして加える、というシンプルな手法で性能を改善。モデルアーキテクチャに依存せず使えるとしている。
翻訳、要約、事前学習モデルの fine-tuning, 言語モデリング, CV など多くのタスクで性能を改善し、正則化の効果を確認している。

![スクリーンショット 2021-07-07 23 59 30](https://user-images.githubusercontent.com/1255485/124782825-83341000-df7f-11eb-8156-a7ab6d7ea6e3.png)

## 2. 先行研究と比べてどこがすごい？

シンプルな修正 (KLDによる損失の追加) でモデルの汎化を大きく改善している。
追加のパラメータなどは必要ない。

## 3. 技術や手法のキモはどこ？

- 通常の分類問題を想定し、x を入力、y をラベルとすると NLL によるロスは
    - ![スクリーンショット 2021-07-08 0 02 23](https://user-images.githubusercontent.com/1255485/124783160-d312d700-df7f-11eb-9d18-af9b58556d03.png)
- これに、同じ入力を2回モデルに通したそれぞれの出力間の KL Divergence を加える
    - ![スクリーンショット 2021-07-08 0 02 33](https://user-images.githubusercontent.com/1255485/124783291-eb82f180-df7f-11eb-8d56-eb3571f3a5f9.png)
    - ![スクリーンショット 2021-07-08 0 03 51](https://user-images.githubusercontent.com/1255485/124783405-ffc6ee80-df7f-11eb-8bc4-94add5bfea34.png)
    - αはハイパーパラメータ。翻訳タスクでは α=5.0, 他のタスクではだいたい 0.1~1.0 程度が与えられていた。
- モデルに2回通す分スピードは落ちるが、2回を batch size を2倍にして1回にまとめることで省力化
- 回帰の場合は KLD の代わりに MSE を使う

## 4. どうやって有効だと検証した？

翻訳から GLUE, CV など幅広く実験し、ほとんど全てで性能を改善している。

翻訳
![スクリーンショット 2021-07-08 0 08 15](https://user-images.githubusercontent.com/1255485/124784157-a1e6d680-df80-11eb-9f70-9a6c8a2f4a1c.png)

GLUE
![スクリーンショット 2021-07-08 0 08 21](https://user-images.githubusercontent.com/1255485/124784179-a7442100-df80-11eb-9a5e-e6effed263fa.png)

## 5. 議論はある？

- pre-training にも使えるかどうか (future work)
- CNN など別アーキテクチャのモデルにも使えるか (future work)
